controllers/common.go

```diff
@@ -13,12 +34,404 @@ func getServiceAccountName() string {
+			memory = *instance.Spec.Deployments.Collector.Resources.Requests.Memory()
+		}
+	case indexerDeploymentName:
+		if instance.Spec.Deployments.Indexer.Resources.Requests.Cpu() != nil {
+			cpu = *instance.Spec.Deployments.Indexer.Resources.Requests.Cpu()
+		}
+		if instance.Spec.Deployments.Indexer.Resources.Requests.Memory() != nil {
+			memory = *instance.Spec.Deployments.Indexer.Resources.Requests.Memory()
+		}
+
+	case postgresDeploymentName:
+		if instance.Spec.Deployments.Database.Resources.Requests.Cpu() != nil {
+			cpu = *instance.Spec.Deployments.Database.Resources.Requests.Cpu()
+		}
+		if instance.Spec.Deployments.Database.Resources.Requests.Memory() != nil {
+			memory = *instance.Spec.Deployments.Database.Resources.Requests.Memory()
+		}
+	}
+
+	return corev1.ResourceList{
+		corev1.ResourceCPU:    cpu,
+		corev1.ResourceMemory: memory,
+	}
+}
+
+func getLimits(deployment string, instance *searchv1alpha1.Search) corev1.ResourceList {
+	var cpu, memory resource.Quantity
+	memory = resource.MustParse(defaultResoureMap[deployment]["MemoryLimit"])
+	if !isResourcesCustomized(deployment, instance) {
+		return corev1.ResourceList{
+			corev1.ResourceMemory: memory,
+		}
+	}
+	switch deployment {
+	case apiDeploymentName:
+		if instance.Spec.Deployments.API.Resources.Requests.Cpu() != nil {
+			cpu = *instance.Spec.Deployments.API.Resources.Requests.Cpu()
+		}
+		if instance.Spec.Deployments.API.Resources.Requests.Memory() != nil {
+			memory = *instance.Spec.Deployments.API.Resources.Requests.Memory()
+		}
+	case collectorDeploymentName:
+		if instance.Spec.Deployments.Collector.Resources.Requests.Cpu() != nil {
+			cpu = *instance.Spec.Deployments.Collector.Resources.Requests.Cpu()
+		}
+		if instance.Spec.Deployments.Collector.Resources.Requests.Memory() != nil {
+			memory = *instance.Spec.Deployments.Collector.Resources.Requests.Memory()
+		}
+	case indexerDeploymentName:
+		if instance.Spec.Deployments.Indexer.Resources.Requests.Cpu() != nil {
+			cpu = *instance.Spec.Deployments.Indexer.Resources.Requests.Cpu()
+		}
+		if instance.Spec.Deployments.Indexer.Resources.Requests.Memory() != nil {
+			memory = *instance.Spec.Deployments.Indexer.Resources.Requests.Memory()
+		}
+
+	case postgresDeploymentName:
+		if instance.Spec.Deployments.Database.Resources.Requests.Cpu() != nil {
+			cpu = *instance.Spec.Deployments.Database.Resources.Requests.Cpu()
+		}
+		if instance.Spec.Deployments.Database.Resources.Requests.Memory() != nil {
+			memory = *instance.Spec.Deployments.Database.Resources.Requests.Memory()
+		}
+	}
+	if cpu.String() == "<nil>" {
+		return corev1.ResourceList{
+			corev1.ResourceMemory: memory,
+		}
+	}
+	return corev1.ResourceList{
+		corev1.ResourceCPU:    cpu,
+		corev1.ResourceMemory: memory,
+	}
+}
+
+func getReplicaCount(deploymentName string, instance *searchv1alpha1.Search) *int32 {
+
+	count := int32(1)
+	if c, ok := defaultReplicaMap[deploymentName]; ok {
+		count = c
+	}
```

Comment:
Does this account for `AvailabilityMode (High or Basic)`?

Reply:
AvailabilityMode - is not accounted in this PR .